src/test/groovy/graphql/execution/DelegatingDataFetcherExceptionHandlerTest.groovy

```diff
@@ -0,0 +1,197 @@
+        delegates.put({ t -> false } as Predicate, Mock(DataFetcherExceptionHandler))
+        def handler = new DelegatingDataFetcherExceptionHandler(delegates)
+
+        when: 'set null defaultHandler'
+        handler.setDefaultHandler(null)
+
+        then: 'fail assert on validation'
+        thrown AssertException
+    }
+
+    def "delegates only to matching delegate"() {
+        given: 'a delegate that matches and a delegate that does not match'
+        def matchingDelegate = Mock(DataFetcherExceptionHandler)
+        def notMatchingDelegate = Mock(DataFetcherExceptionHandler)
+        def expectedResult = Mock(DataFetcherExceptionHandlerResult)
+        def delegates = new LinkedHashMap<Predicate<Throwable>, DataFetcherExceptionHandler>()
+        delegates.put({ t -> true } as Predicate, matchingDelegate)
+        delegates.put({ t -> false } as Predicate, notMatchingDelegate)
+        def environment = DataFetchingEnvironmentImpl.newDataFetchingEnvironment()
+                .build()
+        def handlerParameters = DataFetcherExceptionHandlerParameters.newExceptionParameters()
+                .dataFetchingEnvironment(environment)
+                .exception(new RuntimeException())
+                .build();
+        def handler = new DelegatingDataFetcherExceptionHandler(delegates)
+
+        when: 'onException invoked'
+        def actualResult = handler.onException(handlerParameters)
+
+        then: 'only the matching delegate is invoked'
+        1 * matchingDelegate.onException(handlerParameters) >> expectedResult
+        0 * _
+
+        and: 'the actual result is equal to the expected result'
+        actualResult == expectedResult
+    }
+
+    def "default used"() {
+        given: 'a delegates that do not match'
+        def location = new SourceLocation(6, 9)
+        def field = newMergedField(newField("f").sourceLocation(location).build()).build()
+        def stepInfo = newExecutionStepInfo().path(ResultPath.fromList(["a", "b"])).type(GraphQLString).build()
+        def notMatchingDelegate = Mock(DataFetcherExceptionHandler)
+        def delegates = new LinkedHashMap<Predicate<Throwable>, DataFetcherExceptionHandler>()
+        delegates.put({ t -> false } as Predicate, notMatchingDelegate)
+        def environment = DataFetchingEnvironmentImpl.newDataFetchingEnvironment()
+                .mergedField(field)
+                .executionStepInfo(stepInfo)
+                .build()
+        def handlerParameters = DataFetcherExceptionHandlerParameters.newExceptionParameters()
+                .dataFetchingEnvironment(environment)
+                .exception(new RuntimeException())
+                .build();
+        def handler = new DelegatingDataFetcherExceptionHandler(delegates)
+
+        when: 'onException invoked'
+        def actualResult = handler.onException(handlerParameters)
+
+        then: 'only the matching delegate is invoked'
+        0 * _
```

Comment:
I am not sure this is testing anything??

Reply:
I think `then` needs updated to be `none of the delegates are invoked`. This is verify that if none of the `delegates` match, then none of them are used (the default should be used).

Reply:
I've updated `then` description to be accurate.